src/unix/fs.c

```diff
@@ -1395,7 +1395,7 @@ static int uv__fs_stat(const char *path, uv_stat_t *buf) {
   int ret;
 
   ret = uv__fs_statx(-1, path, /* is_fstat */ 0, /* is_lstat */ 0, buf);
-  if (ret != UV_ENOSYS)
+  if (ret != UV_ENOSYS && ret != UV_EPERM && ret != UV_EACCES)
```

Comment:
statx() can legitimately return EPERM. I don't like the idea of introducing a performance regression (making two system calls where one would suffice) if it's just to work around a buggy seccomp filter.

Reply:
Would you be okay with just doing the fallback for `EPERM`?

Reply:
Yes, that seems fine to me.

Reply:
I've updated the PR to only fall back for `EPERM` (and of course `ENOSYS`).

Reply:
> statx() can legitimately return EPERM EACCES. I don't like the idea of introducing a performance regression (making two system calls where one would suffice) if it's just to work around a buggy seccomp filter.

We rejected this in our fork for the same reason. The reason this PR appears to work is that `uv__fs_statx` returns either `UV_ENOSYS` or `-1` on failure. Now `-1` happens to be `UV_PERM` on linux, but that's basically a coincidence. The workaround for an actual `EPERM` is already inside `uv__fs_statx`. Ideally `uv__fs_statx` would be cleaned up to not be confusing about whether it returns an error code or `-1`, but in any case, this PR doesn't do what one would naively assume it does.